### PR TITLE
[J9] consume jdt version in maven/gradle that does support Java 9

### PR DESCRIPTION
**Pull request by cdietrich**
[J9] consume jdt version in maven/gradle that does support Java 9
https://github.com/eclipse/xtext-extras/issues/211

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.builder.standalone/build.gradle [hunk 1]
```diff
@@ -6,4 +6,5 @@ dependencies {
 	compile project(':org.eclipse.xtext.common.types')
 	compile "org.eclipse.jdt:org.eclipse.jdt.core:$versions.jdtCore"
 	compile "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerApt"
+    compile "org.eclipse.jdt:org.eclipse.jdt.compiler.tool:$versions.jdtCompilerTool"
```
Comment (szarnekow) [inline org.eclipse.xtext.builder.standalone/build.gradle]: Stylistic: Mixed indentation

File: gradle/versions.gradle [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+
+version = '2.14.0-SNAPSHOT'
+
+ext.versions = [
+	'xtext': version,
+	'xtext_bootstrap': '2.13.0',
+	'gradle_plugins': '0.1.0',
+	'xtext_gradle_plugin': '1.0.20',
+	'emfEcore': '[2.10.2,3)',
+	'emfMwe': '1.3.21.201705291010',
```
Comment (szarnekow) [inline gradle/versions.gradle]: Do we need an MWE version with asm 6 here, too?

File: org.eclipse.xtext.generator/build.gradle [hunk 3]
```diff
@@ -9,6 +9,7 @@ dependencies {
 	compile "org.eclipse.xpand:org.eclipse.xtend:$versions.xpand"
 	optional "org.eclipse.jdt:org.eclipse.jdt.core:$versions.jdtCore"
 	optional "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerApt"
+    optional "org.eclipse.jdt:org.eclipse.jdt.compiler.tool:$versions.jdtCompilerTool"
```
Comment (szarnekow) [inline org.eclipse.xtext.generator/build.gradle]: Mixed indentation

File: org.eclipse.xtext.java/build.gradle [hunk 4]
```diff
@@ -5,4 +5,5 @@ dependencies {
 	compile project(':org.eclipse.xtext.common.types')
 	compile "org.eclipse.jdt:org.eclipse.jdt.core:$versions.jdtCore"
 	compile "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerApt"
+    compile "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerTool"
```
Comment (szarnekow) [inline org.eclipse.xtext.java/build.gradle]: Same

File: org.eclipse.xtext.purexbase.tests/build.gradle [hunk 5]
```diff
@@ -4,6 +4,7 @@ dependencies {
 	compile project(':org.eclipse.xtext.xbase.testdata')
 	compile "org.eclipse.jdt:org.eclipse.jdt.core:$versions.jdtCore"
 	compile "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerApt"
+    compile "org.eclipse.jdt:org.eclipse.jdt.compiler.tool:$versions.jdtCompilerTool"
```
Comment (szarnekow) [inline org.eclipse.xtext.purexbase.tests/build.gradle]: Same

File: org.eclipse.xtext.xbase.testing/build.gradle [hunk 6]
```diff
@@ -7,6 +7,7 @@ dependencies {
 	compile "junit:junit:$versions.junit"
 	compile "org.eclipse.jdt:org.eclipse.jdt.core:$versions.jdtCore"
 	compile "org.eclipse.jdt:org.eclipse.jdt.compiler.apt:$versions.jdtCompilerApt"
+    compile "org.eclipse.jdt:org.eclipse.jdt.compiler.tool:$versions.jdtCompilerTool"
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.testing/build.gradle]: Same

File: gradle/versions.gradle (hunk 2 above)
Comment (cdietrich) [reply gradle/versions.gradle]: that is not yet built :(

**Comment by cdietrich** [comment]
updated

**Pull request merged**